### PR TITLE
Viewcontroller

### PR DESCRIPTION
**Pull request by westnordost**
See #3830

What do you think?

The plan would be to eventually move all common view logic of different quest forms into those "ViewController"s. Of course it is not a 100% replacement for the A*SomethingSomething*Fragment but most of it.

### CODE REVIEW

File: app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.kt [hunk 1]
```diff
@@ -97,30 +56,12 @@ class AddMaxHeightForm : AbstractQuestFormAnswerFragment<MaxHeightAnswer>() {
     }
 
     private fun userSelectedUnrealisticHeight(): Boolean {
-        val height = getHeightFromInput() ?: return false
-        val m = height.toMeters()
-        return m > 6 || m < 1.9
+        val m = lengthInput.length?.toMeters() ?: return false
+        return m > 6 || m < 1.8
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.kt:60]: Is the change from `1.9` to `1.8` deliberate?

File: app/src/main/java/de/westnordost/streetcomplete/quests/width/AddWidthForm.kt [hunk 2]
```diff
@@ -34,10 +36,15 @@ class AddWidthForm : AbstractQuestFormAnswerFragment<WidthAnswer>() {
         binding.widthExplanationTextView.isGone = explanation == null
         binding.widthExplanationTextView.text = explanation
 
-        binding.lengthInput.maxFeetDigits = if (isRoad) 3 else 2
-        binding.lengthInput.maxMeterDigitsBeforeDecimalPoint = if (isRoad) 2 else 1
-        binding.lengthInput.selectableUnits = countryInfo.lengthUnits
-        binding.lengthInput.onInputChanged = {
+        lengthInput = binding.lengthInput.let {
+            LengthInputViewController(it.unitSelect, it.metersContainer, it.metersInput, it.feetInchesContainer, it.feetInput, it.inchesInput, )
```
Comment (FloEdelmann) [inline app/src/main/java/de/westnordost/streetcomplete/quests/width/AddWidthForm.kt:40]:
```suggestion
            LengthInputViewController(it.unitSelect, it.metersContainer, it.metersInput, it.feetInchesContainer, it.feetInput, it.inchesInput)
```

**Review by FloEdelmann — COMMENTED** [review]
Looks good! (just from a code review on GitHub)

File: app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.kt (hunk 1 above)
Comment (westnordost) [reply app/src/main/java/de/westnordost/streetcomplete/quests/max_height/AddMaxHeightForm.kt:60]: yes

**Comment by westnordost** [comment]
Thanks for the review!
I meant also in general, the pattern and the naming thereof.

**Comment by FloEdelmann** [comment]
I like the pattern, but I couldn't really think of (or find) a better name.

**Pull request merged**